From 14b2f735c1b836cebcf7277585d7821a416f43ba Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Tue, 22 Mar 2016 16:56:39 +0800
Subject: [PATCH 0309/2241] arm64: mmc: armada3700: move delay_adjust_card
 definition

Move delay_adjust_card from card_cntx to struct
sdhci_xenon_priv, to avoid conflict with Linux 4.4
which uses handler_priv for MMC GPIO.
Remove struct card_cntx and corresponding operations.

Change-Id: I791df9bc53e5bda7927a758c786e492e5c810555
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28425
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Shadi Ammouri <shadi@marvell.com>
---
 drivers/mmc/host/sdhci-xenon.c | 20 +++++---------------
 1 file changed, 5 insertions(+), 15 deletions(-)

diff --git a/drivers/mmc/host/sdhci-xenon.c b/drivers/mmc/host/sdhci-xenon.c
index ab7fe62..a4ddf11 100644
--- a/drivers/mmc/host/sdhci-xenon.c
+++ b/drivers/mmc/host/sdhci-xenon.c
@@ -197,10 +197,7 @@ struct sdhci_xenon_priv {
 	unsigned char	timing;
 	unsigned char	tuning_count;
 	unsigned int	clock;
-	unsigned int	quirks; /* Xenon private quirks */
-};
 
-struct card_cntx {
 	/* When initializing card, Xenon has to adjust
 	 * sampling fixed delay.
 	 * However, at that time, card structure is not
@@ -210,6 +207,8 @@ struct card_cntx {
 	 * of the card during initialization
 	 */
 	struct mmc_card *delay_adjust_card;
+
+	unsigned int	quirks; /* Xenon private quirks */
 };
 
 /*
@@ -683,7 +682,6 @@ static int sdhci_xenon_delay_adj(struct sdhci_host *host, struct mmc_ios *ios)
 	struct mmc_card *card = NULL;
 	struct sdhci_pltfm_host *pltfm_host = sdhci_priv(host);
 	struct sdhci_xenon_priv *priv;
-	struct card_cntx *cntx;
 
 	if (!host->clock)
 		return 0;
@@ -715,8 +713,7 @@ static int sdhci_xenon_delay_adj(struct sdhci_host *host, struct mmc_ios *ios)
 	priv->bus_width = ios->bus_width;
 	priv->timing = ios->timing;
 
-	cntx = (struct card_cntx *)mmc->slot.handler_priv;
-	card = cntx->delay_adjust_card;
+	card = priv->delay_adjust_card;
 	if (unlikely(card == NULL))
 		return -EIO;
 
@@ -760,11 +757,10 @@ static void sdhci_xenon_init_card(struct sdhci_host *host,
 {
 	u32 reg;
 	u8 slot_idx;
-	struct card_cntx *cntx;
 	struct mmc_host *mmc = host->mmc;
+	struct sdhci_xenon_priv *priv = sdhci_priv(host);
 
-	cntx = (struct card_cntx *)mmc->slot.handler_priv;
-	cntx->delay_adjust_card = card;
+	priv->delay_adjust_card = card;
 
 	slot_idx = mmc->slotno;
 
@@ -1147,12 +1143,6 @@ static int sdhci_xenon_slot_probe(struct sdhci_host *host)
 	struct sdhci_pltfm_host *pltfm_host = sdhci_priv(host);
 	struct sdhci_xenon_priv *priv =
 				 (struct sdhci_xenon_priv *)pltfm_host->private;
-	struct card_cntx *cntx;
-
-	cntx = devm_kzalloc(mmc_dev(mmc), sizeof(*cntx), GFP_KERNEL);
-	if (!cntx)
-		return -ENOMEM;
-	mmc->slot.handler_priv = cntx;
 
 	/* Enable slot */
 	sdhci_xenon_set_slot(host, slotno, true);
-- 
2.7.4

